### ISSUE TITLE
Condition can't be inferred, unable to merge overloads

### ISSUE THREAD

**Issue by daxpryce** [beartype/beartype#112]
So, with 10.3 releasing, our nightly downstream dependency check build failed telling us:

```bash
mypy ./graspologic
/Users/runner/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/beartype/_decor/main.py:294: error: Condition can't be inferred, unable to merge overloads
Found 1 error in 1 file (checked 79 source files)
```

Mypy coming in with the best error messages that tell me nothing actionable there, but it at least seems like it's sad at something in the new beartype?  Any ideas @leycec?  It's entirely too Friday to have to deal with this shit, so I'm sorry about that!

https://github.com/microsoft/graspologic/runs/5513695671?check_suite_focus=true#step:6:1 - for if you would like to inspect any other of the entirely overcomplicated CICD build process that may have some impact on this

**Comment by twoertwein** [comment]
Might be a mypy 0.940 issue. When I downgrade mypy to 0.931 I do not get this error.

**Comment by twoertwein** [comment]
I think this issue can be closed, see https://github.com/python/mypy/issues/12335#issuecomment-1065591703.

@leycec It might be worth to have a quick release with the workaround suggested by @cdce8p (see the linked comment).

**Comment by leycec** [comment]
**ohnoes.** The inglorious `0.10.x` dev cycle continues with an unexpected free bonus epilogue chapter. Thanks so much for the rapid response time, @daxpryce and @twoertwein. The direct link to @cdce8p's [proposed zero-day hot fix](https://github.com/python/mypy/issues/12335#issuecomment-1065591703) was especially helpful, @twoertwein.

`beartype 0.10.4` be like:

<img src="https://i.giphy.com/media/l1uguw2lgyys47MxW/100.webp" width="1024">

**Comment by leycec** [comment]
**Resolved by ccac042.** It turns out that if you don't sleep, you can solve problems. Tests pass. Therefore, we pass out. :sleeping: 

Oh – and we're dropping the `beartype 0.10.4` ~~bomb~~ bundle of joy "soon." Soon probably means Sunday or Monday evening. (On Saturdays, I pretend none of this is happening.)
